openeo-geotrellis/src/main/scala/org/openeo/geotrellis/layers/DefaultItemRasterSourceProvider.scala

```diff
@@ -0,0 +1,290 @@
+package org.openeo.geotrellis.layers
+
+import cats.data.NonEmptyList
+import geotrellis.proj4.CRS
+import geotrellis.raster.ResampleMethods.NearestNeighbor
+import geotrellis.raster.gdal.{GDALPath, GDALRasterSource, GDALWarpOptions}
+import geotrellis.raster.geotiff.{GeoTiffPath, GeoTiffRasterSource, GeoTiffReprojectRasterSource, GeoTiffResampleRasterSource}
+import geotrellis.raster.io.geotiff.OverviewStrategy
+import geotrellis.raster.{CellSize, CellType, ConvertTargetCellType, GridBounds, GridExtent, MultibandTile, Raster, RasterExtent, RasterMetadata, RasterSource, ResampleMethod, ResampleTarget, ShortConstantNoDataCellType, SourceName, TargetAlignment, TargetCellType, TargetRegion, UByteUserDefinedNoDataCellType, UShortConstantNoDataCellType}
+import geotrellis.vector.{Extent, ProjectedExtent}
+import org.openeo.geotrellis.file.FixedFeaturesOpenSearchClient
+import org.openeo.geotrellis.layers.FileLayerProvider.{convertNetcdfLinksToGDALFormat, vsis3ToS3}
+import org.openeo.geotrelliscommon.DataCubeParameters
+import org.openeo.opensearch.{OpenSearchClient, OpenSearchResponses}
+import org.openeo.opensearch.OpenSearchResponses.{Feature, Link}
+import org.slf4j.{Logger, LoggerFactory}
+
+import java.net.URI
+import java.nio.file.Paths
+
+case class DefaultItemRasterSourceProvider(openSearch: OpenSearchClient, openSearchLinkTitles: NonEmptyList[String], rootPath: String,
```

Comment:
We probably want to work out the dependency on OpenSearchClient in the long run?
(Somewhat in doubt myself if we should aim to remove it in this PR.)